### ISSUE TITLE
Generate deployment packages

### ISSUE THREAD

**Issue by mmatyas** [mmatyas/pegasus-frontend#15]
- [x] Create a simple TAR package for Linux
  - ~(fails for the same reason, as it also uses the same tool)~
  - fixed by static builds
- [x] Create a simple ZIP package for Windows
- [ ] Add AppImage support
  - previously: *outdated Travis, Docker complications, missing libs: too much hacking. Works offline though. I'll build Qt on older Ubuntu next time*
  - I'd need to bundle GStreamer, is that okay?
- [ ] Snap support
  - there are some Qt parts for Snapcraft, but a lot missing; would require building Qt from scratch
  - system access problems, as listed below
- [ ] Flatpak support
  - it seems I'd need to depend on the whole KDE runtime to get it work
  - maybe I can avoid it with static builds
  - TODO test

**Comment by mmatyas** [comment]
Resolved byy #21

**Comment by mmatyas** [comment]
I've tried Snapcraft again, but had the following issues:

- the `home` plug has no access to hidden directories by design -- it seems I can't use neither `.config` nor `.emulationstation`.
- no OpenGL access despite the `opengl` plug

```
libGL error: unable to load driver: nouveau_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: nouveau
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
Unrecognized OpenGL version
Unrecognized OpenGL version
```

**Comment by mmatyas** [comment]
Closing for now until someone reports problems with the static builds.

**Comment by jackdroido** [comment]
Hi @mmatyas !

First of all, thanks for creating this beautiful frontend.

If you are still considering to support flatpak, I wrote a simple manifest here: https://gist.github.com/jackdroido/0aad70c121f3747b01263541b23727c7

Here's the steps I use for building and installing locally:
`flatpak install --user flathub org.kde.Platform//5.12 org.kde.Sdk//5.12`
`flatpak-builder --repo=repo --force-clean build-dir org.frontend.Pegasus.json`
`flatpak --user remote-add --no-gpg-verify pegasus-repo repo`
`flatpak --user install pegasus-repo org.frontend.Pegasus`

From a quick test, it seems to work fine. Hope it helps to see Pegasus on flathub soon ;-)

**Comment by mmatyas** [comment]
Nice! I'm not against the idea; previously the problem with Flatpak was that only the KDE runtime contained a somewhat up-to-dateQt, but it's size was was in the several hundred megabytes range. I wonder if/how the situation changed since.

**Comment by jackdroido** [comment]
The KDE runtime is not strictly "needed". One could grab, build and embed only the libraries one application depends on (QT in this case) by using a smaller runtime like the Freedesktop one, or go further and create a runtime from scratch.

But IMO this goes a little against the whole idea of Flatpak: having a stable and tested base an application could depend on for easy distribution and maintenance...and without duplication, since a runtime is shared among all flatpaks that use it.

**Comment by mmatyas** [comment]
Just to keep you updated, I've finished writing the necessary flatpak files, will do some more testing then I think I can start the publishing process tomorrow :)

However, note that Flathub's policy is to only hosts stable releases; you will not be able to test the latest updates with this version. In fact, I wonder if they accept Pegasus in the first place...

**Comment by mmatyas** [comment]
It's also seems to be preferred to have the appdata.xml in the main repo, so I think I'll wait with the publishing until the next "stable" release.

**Comment by jackdroido** [comment]
Nice ! In the meanwhile I've updated the gist above, and created a little patch with some little AppStream fixes here: https://gist.github.com/jackdroido/fb815b33b79b0f1bf2c787a95bca5685

Mainly, using /usr/share/metainfo/ instead of /usr/share/appdata/, and removing the `<icon>` tag which is invalid in a desktop-application component and made the build fail.

Source: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent

**Comment by mmatyas** [comment]
Ah you're right, I didn't notice the path/filename change in the spec, will update. Also thanks for the message fix.

The icon is an interesting thing, as the page you've also linked explicitly says it *is* available for `desktop-application`s. And indeed, if you try to validate it using the official `appstreamcli`, it passes. If you use the GNOME peoples' tool to check it, like what Flatpak does, that is what reports such errors. Anyway, I've removed the icon line, I think that tag's intent isn't what I thought either.

I also have a manifest [here](https://gist.github.com/mmatyas/53a22afb037aea800480d3da7685ecdb) which I've almost published, then I've started playing with the other tags like this `icon`.
